sros2/sros2/api/__init__.py

```diff
@@ -226,12 +247,16 @@ def is_key_name_valid(name):
 
 
 def create_permission_file(path, domain_id, policy_element):
+    print('creating permission')
     permissions_xsl_path = get_transport_template('dds', 'permissions.xsl')
     permissions_xsl = etree.XSLT(etree.parse(permissions_xsl_path))
     permissions_xsd_path = get_transport_schema('dds', 'permissions.xsd')
     permissions_xsd = etree.XMLSchema(etree.parse(permissions_xsd_path))
 
-    permissions_xml = permissions_xsl(policy_element)
+    kwargs = {}
+    if get_current_rmw_implementation() in ('rmw_fastrtps_cpp','rmw_fastrtps_dynamic_cpp'):
+        kwargs['allow_ros_discovery_topic'] = etree.XSLT.strparam("1")
+    permissions_xml = permissions_xsl(policy_element, **kwargs)
```

Comment:
Whoa, that cool! Didn't know xsl templates could take in `kwargs`, TDIL.

Reply:
Me neither :rofl:. I liked it, because it avoids duplicating the template (with minimal changes).